staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -284,12 +288,16 @@ func (pf *PortForwarder) getListener(protocol string, hostname string, port *For
 	localPortUInt, err := strconv.ParseUint(localPort, 10, 16)
 
 	if err != nil {
+		pf.outLock.Lock()
+		defer pf.outLock.Unlock()
```

Comment:
interesting, why this doesn't check that pf.out is not nil?

Reply:
You'll have to ask *checks git blame* Chao Xu from 6 years ago. Might as well add it here too.

Reply:
😄